src/overload_all.jl

```diff
@@ -0,0 +1,30 @@
+function overload_all(M)
+    exprs_1_to_1 = [
+        quote
+            $(overload_connectivity_1_to_1(M, op))
+            $(overload_gradient_1_to_1(M, op))
+            $(overload_hessian_1_to_1(M, op))
+        end for op in nameof.(list_operators_1_to_1(Val(M)))
+    ]
+    exprs_2_to_1 = [
+        quote
+            $(overload_connectivity_2_to_1(M, op))
+            $(overload_gradient_2_to_1(M, op))
+            $(overload_hessian_2_to_1(M, op))
+        end for op in nameof.(list_operators_2_to_1(Val(M)))
+    ]
+    exprs_1_to_2 = [
+        quote
+            $(overload_connectivity_1_to_2(M, op))
+            $(overload_gradient_1_to_2(M, op))
+            $(overload_hessian_1_to_2(M, op))
+        end for op in nameof.(list_operators_1_to_2(Val(M)))
+    ]
+    return quote
+        $(exprs_1_to_1...)
+        $(exprs_2_to_1...)
+        $(exprs_1_to_2...)
+    end
+end
+
+eval(overload_all(:Base))
```

Comment:
Yeah, this combined with `list_operators_*` and the big `quote`s in `overload_*.jl` feels very hacky. 🫤
There are too many layers of metaprogramming with a lot of implicit dependencies in their design that interact across several files.

Reply:
It's exactly the way they do it in ForwardDiff:

- The big quotes: https://github.com/JuliaDiff/ForwardDiff.jl/blob/1907196ba50e3eb337860301ece3ec37d0582257/src/dual.jl#L230-L244
- The evaluation: https://github.com/JuliaDiff/ForwardDiff.jl/blob/1907196ba50e3eb337860301ece3ec37d0582257/src/dual.jl#L469-L472
- The listing: https://github.com/JuliaDiff/ForwardDiff.jl/blob/1907196ba50e3eb337860301ece3ec37d0582257/src/dual.jl#L463

Reply:
I spent quite a lot of time trying out various things to make it work with minimal changes, and this seems to work.

Reply:
What's in the quotes is exactly the same as what was directly `eval`-ed earlied (if you ignore the additional module prefixes, which I got from ForwardDiff).
The difference is that we split code generation (creating the expression) from code evaluation (running it through `eval`)

Reply:
The contents of `overload_all` are only designed so that it can be a one-liner in all the package extensions we will no doubt need to add